RecoCTPPS/TotemRPLocal/src/TotemTimingConversions.cc

```diff
@@ -3,119 +3,157 @@
+    openCalibrationFile();
+  float t = firstCellTimeInstant + parsedData_.getTimeOffset(db, sampic, channel);
+  //NOTE: If no time offset is set, getTimeOffset returns 0
+
+  if (mergeTimePeaks_) {
+    if (t < -ACCEPTED_TIME_RADIUS)
+      t += SAMPIC_MAX_NUMBER_OF_SAMPLES * SAMPIC_SAMPLING_PERIOD_NS;
+    if (t > ACCEPTED_TIME_RADIUS)
+      t -= SAMPIC_MAX_NUMBER_OF_SAMPLES * SAMPIC_SAMPLING_PERIOD_NS;
+  }
+  return t;
 }
 
 //----------------------------------------------------------------------------------------------------
 
-const float TotemTimingConversions::getTriggerTime(const TotemTimingDigi& digi) const
+float
+TotemTimingConversions::getTriggerTime(const TotemTimingDigi& digi)
 {
+  if (!calibrationFileOpened_)
+    openCalibrationFile();
+
   unsigned int offsetOfSamples = digi.getEventInfo().getOffsetOfSamples();
   if (offsetOfSamples == 0)
     offsetOfSamples = 30; // FW 0 is not sending this, FW > 0 yes
 
-  float triggerCellTimeInstant = getTimeOfFirstSample(digi) +
-        (SAMPIC_MAX_NUMBER_OF_SAMPLES - offsetOfSamples) *
-        SAMPIC_SAMPLING_PERIOD_NS;
+  return getTimeOfFirstSample(digi) + (SAMPIC_MAX_NUMBER_OF_SAMPLES - offsetOfSamples) * SAMPIC_SAMPLING_PERIOD_NS;
+}
 
-  return triggerCellTimeInstant;
+//----------------------------------------------------------------------------------------------------
+
+float
+TotemTimingConversions::getTimePrecision(const TotemTimingDigi& digi)
+{
+  if (!calibrationFileOpened_)
+    openCalibrationFile();
+  int db = digi.getHardwareBoardId();
+  int sampic = digi.getHardwareSampicId();
+  int channel = digi.getHardwareChannelId();
+  return parsedData_.getTimePrecision(db, sampic, channel);
 }
 
 //----------------------------------------------------------------------------------------------------
 
-std::vector<float> TotemTimingConversions::getTimeSamples(const TotemTimingDigi& digi) const
+std::vector<float>
+TotemTimingConversions::getTimeSamples(const TotemTimingDigi& digi)
 {
+  if (!calibrationFileOpened_)
+    openCalibrationFile();
+
   std::vector<float> time(digi.getNumberOfSamples());
-  // firstCellTimeInstant = 0;
-  if (!calibrationFileOk_)
-  {
-    for (unsigned int i = 0; i < time.size(); ++i)
-      time.at(i) = getTimeOfFirstSample(digi) + i * SAMPIC_SAMPLING_PERIOD_NS;
-  }
+  for (unsigned int i = 0; i < time.size(); ++i)
+    time.at(i) = getTimeOfFirstSample(digi) + i * SAMPIC_SAMPLING_PERIOD_NS;
   return time;
 }
 
 //----------------------------------------------------------------------------------------------------
+// NOTE: If no proper file is specified, calibration is not applied
 
-std::vector<float> TotemTimingConversions::getVoltSamples(const TotemTimingDigi& digi) const
+std::vector<float>
+TotemTimingConversions::getVoltSamples(const TotemTimingDigi& digi)
 {
+  if (!calibrationFileOpened_)
+    openCalibrationFile();
+
   std::vector<float> data;
-  if (!calibrationFileOk_)
-  {
+  if (calibrationFile_.empty())
```

Comment:
perhaps a more explicit flag ```ignoreCalibration_``` or similar is more efficient that checking a string.
As mentioned in the comments to the .h file, I think that the setup should happen once per TotemTimingConversions creation and not passed to the first possible caller

Reply:
Indeed! now making use of the previously underrated `calibrationFileOpened_` class member.